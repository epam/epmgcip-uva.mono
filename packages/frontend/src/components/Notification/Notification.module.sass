@use 'src/colors'
@import 'src/typography'

.notificationWrapper
    @extend .H6
    z-index: -1
    position: absolute
    top: 20px
    left: 50%
    transform: translateX(-50%)
    opacity: 0
    min-width: 200px
    width: 90%
    max-width: 768px
    min-height: 80px
    border: 1px solid colors.$blue-dark
    background-color: colors.$white
    border-radius: 8px
    padding: 12px
    box-sizing: border-box
    display: flex
    justify-content: center
    align-items: center
    text-align: justify
    box-shadow: 0px 5px 5px 0px rgba(0, 0, 0, 0.5)