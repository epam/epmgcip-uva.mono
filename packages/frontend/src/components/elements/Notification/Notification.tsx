import { DEFAULT_NOTIFICATION } from 'src/constants';
import css from './Notification.module.sass';
import { hideElement } from 'src/utils/hideElement';

export const Notification = () => {
  const notification = document.getElementById('notification');
  const handleHideNotification = () => {
    if (notification) {
      hideElement(notification);
    }
  };

  return (
    <div id='notification' data-testid = 'notification-test-id' className={css.notificationWrapper} onClick={handleHideNotification}>
      {DEFAULT_NOTIFICATION}
    </div>
  );
};
