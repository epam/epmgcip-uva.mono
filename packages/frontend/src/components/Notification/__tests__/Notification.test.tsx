import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Notification } from '../Notification';

describe('Testing: Notification', () => {
  it('should render', () => {
    const { container } = render(<Notification />);

    expect(container).toMatchSnapshot();
  });
});
