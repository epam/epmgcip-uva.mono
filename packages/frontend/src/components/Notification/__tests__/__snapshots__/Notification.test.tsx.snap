// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Notification > should render 1`] = `
<div>
  <div
    class="_notificationWrapper_3741e4"
    id="notification"
  >
    Default Notification Message
  </div>
</div>
`;
